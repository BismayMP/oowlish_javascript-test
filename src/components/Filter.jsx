import React, { useState } from 'react';
import FilterFields from './FilterFields';

const Filter = ({ table }) => {
    const [state, setState] = useState({ name: '', UPIN: '' });

    const handleOnFilterChange = (event) => {
        const { name, value } = event.target;
        setState({ ...state, [name]: value.toLowerCase() })
    }

    const applyFilterToTable = () => {
        try {
            for (let tr of table) {
                (`${tr.getAttribute('data-upin')}`.includes(state.UPIN) &&
                    tr.cells[0].textContent.toLowerCase().includes(state.name))
                    ? tr.removeAttribute('hidden') : tr.setAttribute('hidden', "")
    
            }
        } catch (error) {
            return error.message
        }
        
    }

    return (
        <>
            <FilterFields
                name={state.name}
                UPIN={state.UPIN}
                onFilterChange={handleOnFilterChange}
            />

            {
                applyFilterToTable()
            }
        </>
    )
}

export default Filter