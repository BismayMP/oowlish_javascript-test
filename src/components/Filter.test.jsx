import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Filter from './Filter'

jest.mock('./FilterFields', () => 'FilterFields')


test('component render', () => {
    const tree = renderer.create(<Filter></Filter>).toJSON()
    expect(tree).toMatchSnapshot()

})
