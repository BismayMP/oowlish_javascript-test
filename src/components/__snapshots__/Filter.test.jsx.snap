// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component render 1`] = `
Array [
  <FilterFields
    UPIN=""
    name=""
    onFilterChange={[Function]}
  />,
  "undefined is not iterable (cannot read property Symbol(Symbol.iterator))",
]
`;
